tests/components/ps4/test_config_flow.py

```diff
@@ -293,6 +299,27 @@
     assert len(manager.async_entries()) == 2
 
 
+async def test_0_pin(hass):
+    """Test Pin leading with 0."""
+    flow = ps4.PlayStation4FlowHandler()
+    flow.hass = hass
+    flow.location = MOCK_LOCATION
+    flow.creds = MOCK_CREDS
+
+    mock_config = MOCK_CONFIG
+    mock_config[CONF_CODE] = MOCK_CODE_LEAD_0
+
+    with patch(
+        "pyps4_2ndscreen.Helper.link", return_value=(True, True)
+    ) as mock_call, patch(
+        "pyps4_2ndscreen.Helper.has_devices", return_value=[{"host-ip": MOCK_HOST}]
+    ):
+        await flow.async_step_link(mock_config)
```

Comment:
Start the flow with the core interface. Here's an example:
https://github.com/home-assistant/home-assistant/blob/52c1bc9c26f7d936e2dcc3c1c70ce98172af31ba/tests/components/hue/test_config_flow.py#L395-L397

Adjust the context to start the correct step.

To continue a flow, use this:
https://github.com/home-assistant/home-assistant/blob/52c1bc9c26f7d936e2dcc3c1c70ce98172af31ba/tests/components/hue/test_config_flow.py#L438

See the hue config flow tests for good examples.